cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -567,13 +580,10 @@ def _get_volume_stats(self):
                         'connection_capabilities': connection_capabilities,
                         'backend_state': backend_state,
                         'pool_state': pool_state,
-                        'pool_down_reason': pool_down_reason
+                        'pool_down_reason': pool_down_reason,
+                        'custom_attributes': custom_attributes,
                         }
 
-                # Add any custom attributes associated with the datastore
-                if "custom_attributes" in datastore:
-                    pool['custom_attributes'] = datastore['custom_attributes']
-
```

Comment:
I'd either:
- keep the lower `if...` to only set the `'custom_attributes'` key conditionally
- or change the dict value to:

```python
   'custom_attributes': custom_attributes or {},
```

so that any subsequent code that may _only_ check for the presence of the `custom_attributes` _key_ and then assume the _value_ to be well-formed doesn't break if `datastore['custom_attributes']` is falsy.